lib/ecto_psql_extras/diagnose_logic.ex

```diff
@@ -0,0 +1,222 @@
+defmodule EctoPSQLExtras.DiagnoseLogic do
+  @moduledoc """
+  Diagnose report logic
+  """
+
+  @table_cache_hit_min_expected 0.985
+  @index_cache_hit_min_expected "0.985"
+  @unused_indexes_max_scans 20
+  @unused_indexes_min_size_bytes 1000000
+  @null_indexes_min_size_mb 1 # 1 MB
+  @null_min_null_frac_percent 50 # 50%
+  @bloat_min_value 10.0
+  @outliers_min_exec_ratio 33 # 33%
+
+  def run(repo) do
+    %{
+      columns: ["ok", "check_name", "message"],
+      rows: [
+        table_cache_hit(repo),
+        index_cache_hit(repo),
+        unused_indexes(repo),
+        null_indexes(repo),
+        bloat(repo),
+        duplicate_indexes(repo),
+        outliers(repo),
+        ssl_used(repo),
+      ]
+    }
+  end
+
+  defp table_cache_hit(repo) do
+    table_cache_hit_val = EctoPSQLExtras.cache_hit(repo, format: :raw).rows
+    |> Enum.at(1) |> Enum.at(1)
+
+    [ok, message] = case table_cache_hit_val do
+      nil ->
+        [false, "Table cache hit ratio is not yet reported."]
+      %Decimal{} = val ->
+        val = Decimal.to_float(val) |> Float.round(6)
+
+        if (val >= @table_cache_hit_min_expected ) do
+          [true, "Table cache hit ratio is correct: #{Float.round(val, 5)}"]
+        else
+          [false, "Table cache hit ratio is too low: #{Float.round(val, 5)}"]
+        end
+    end
+
+    [ok, "table_cache_hit", message]
+  end
+
+  defp index_cache_hit(repo) do
+    index_cache_hit_val = EctoPSQLExtras.cache_hit(repo, format: :raw).rows
+    |> Enum.at(0) |> Enum.at(1)
+
+    [ok, message] = case index_cache_hit_val do
+      nil ->
+        [false, "Index cache hit ratio is not yet reported."]
+      %Decimal{} = val ->
+        val = Decimal.to_float(val) |> Float.round(6)
+
+        if (val >= @index_cache_hit_min_expected ) do
```

Comment:
You don't need the `( )` here:
```suggestion
        if val >= @index_cache_hit_min_expected do
```